### PR TITLE
Fix/1474 bencedcleth

### PR DESCRIPTION
**Pull request by NickKhalow**
## What does this PR change?

Fixes dereferencing for the parent on a child deletion
...

## How to test the changes?

Follow the issue: https://github.com/decentraland/unity-explorer/issues/1474

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Comment by NickKhalow** [comment]
I caught this issue too, it also appears in the mini game itself

image

File: Explorer/Assets/DCL/Rendering/Highlight/HighlightedObject/HighlightedObjects.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                HighlightSettings highlightSettings = highLightRenderers[renderer];
+                highlightSettings.Color = color;
+                highlightSettings.Width = thickness;
+                highLightRenderers[renderer] = highlightSettings;
+            }
+        }
+
+        public void Disparage(Renderer renderer)
+        {
+            if (highLightRenderers.ContainsKey(renderer))
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/Rendering/Highlight/HighlightedObject/HighlightedObjects.cs:36]: Checking for a key before removal is redundant

File: Explorer/Assets/DCL/Rendering/Highlight/HighlightedObject/HighlightedObjects.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        private readonly Dictionary<Renderer, HighlightSettings> highLightRenderers;
+
+        public HighlightedObjects(Dictionary<Renderer, HighlightSettings> highLightRenderers)
+        {
+            this.highLightRenderers = highLightRenderers;
+        }
+
+        public void Highlight(Renderer renderer, Color color, float thickness)
+        {
+            if (!highLightRenderers.ContainsKey(renderer))
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/Rendering/Highlight/HighlightedObject/HighlightedObjects.cs:17]: You could merge `ContainsKey` and `[renderer]` into `TryGet`

**Review by mikhail-dcl — APPROVED** [review]
I just have minor suggestions

File: Explorer/Assets/DCL/Rendering/Highlight/HighlightedObject/HighlightedObjects.cs (hunk 1 above)
Comment (NickKhalow) [reply Explorer/Assets/DCL/Rendering/Highlight/HighlightedObject/HighlightedObjects.cs:37]: Thanks for the notation, I simplified it even more, before I moved the original logic from the system

**Review by mikhail-dcl — APPROVED** [review]

**Pull request merged**